tst/unit/test_swarm.cpp

```diff
@@ -61,10 +82,16 @@ TEST_CASE("Swarm memory management", "[Swarm]") {
   Packages_t packages;
   auto meshblock = std::make_shared<MeshBlock>(1, 1);
   auto mesh = std::make_shared<Mesh>(pin.get(), app_in.get(), packages, 1);
+  mesh->mesh_bcs[0] = BoundaryFlag::user;
+  mesh->SwarmBndryFnctn[0] = SetSwarmIX1UserBC;
+  for (int i = 1; i < 6; i++) {
+    mesh->mesh_bcs[i] = BoundaryFlag::outflow;
+  }
   meshblock->pmy_mesh = mesh.get();
   Metadata m;
   auto swarm = std::make_shared<Swarm>("test swarm", m, NUMINIT);
   swarm->SetBlockPointer(meshblock);
+  swarm->AllocateBoundaries();
   auto swarm_d = swarm->GetDeviceContext();
```

Comment:
This change here, just checks that the user boundaries can be allocated, correct?
Otherwise, I'm not seeing where the boundary function actually becomes active, i.e., the test case where a particles leaves the domain in this unit test.

Reply:
Yes that's right. I was leveraging the `particles` example (which isn't even part of the CI) but that defaults to periodic BCs anyway. I just updated the swarm unit test to move a particle below the `IX1` boundary and use the custom boundary condition to mark it for removal and then remove it and then make sure it was removed.